Can't install on linux (codeship CI)

Hi, I'm trying to use @sentry/cli through @sentry/webpack-plugin, but it won't install on Codeship CI.

It installs on my OSX

Running `yarn install` on the CI server yields the following:

```text
error /home/path/to/frontend/node_modules/@sentry/cli: Command failed.
Exit code: 1
Command: node scripts/install.js
Arguments: 
Directory: /home/path/to/frontend/node_modules/@sentry/cli
Output:
fetching sentry-cli-Linux-x86_64 1%
[...]
fetching sentry-cli-Linux-x86_64 100%
Error: spawn /home/path/to/frontend/node_modules/@sentry/cli/js/__mocks__/sentry-cli ENOENT
```

I've also tried with `npm install` but it yields a similar error:

```text
> @sentry/cli@1.29.1 install /home/rof/src/github.com/DoveConviene/s2smanager/frontend/node_modules/@sentry/cli
> node scripts/install.js

fetching sentry-cli-Linux-x86_64 ████████████████████ 100% 0.0s
Error: spawn /home/path/to/frontend/node_modules/@sentry/cli/js/__mocks__/sentry-cli ENOENT

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @sentry/cli@1.29.1 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @sentry/cli@1.29.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/rof/.npm/_logs/2018-02-13T14_07_49_470Z-debug.log
```

I've opened the log file mentioned by npm and here are the relevant lines:

```text
50939 silly install @sentry/cli@1.29.1
50940 info lifecycle @sentry/cli@1.29.1~install: @sentry/cli@1.29.1
50941 verbose lifecycle @sentry/cli@1.29.1~install: unsafe-perm in lifecycle true
50942 verbose lifecycle @sentry/cli@1.29.1~install: PATH: /home/rof/.nvm/versions/node/v8.9.4/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/path/to/frontend/node_modules/@sentry/cli/node_modules/.bin:/home/path/to/frontend/node_modules/.bin:/home/rof/.nvm/versions/node/v8.9.4/bin:/home/rof/.yarn/bin:/home/rof/.rvm/gems/ruby-2.1.7/bin:/home/rof/.rvm/gems/ruby-2.1.7@global/bin:/home/rof/.rvm/rubies/ruby-2.1.7/bin:node_modules/.bin:/home/rof/.packer:/home/rof/dart-sdk/bin:/home/rof/google-cloud-sdk/bin:/home/rof/appengine/python_appengine:/home/rof/appengine/go_appengine:/home/rof/appengine/java_appengine/bin:/home/rof/cache/composer/bin:/home/rof/cache/bundler/bin:/home/rof/firefox:/home/rof/.slimerjs:/home/rof/.phpenv/bin:/home/rof/.phpenv/shims:/home/rof/.pyenv/bin:/home/rof/.pyenv/shims:/home/rof/bin:/home/rof/.gradle/gradle/bin:/home/rof/.ant/ant/bin:/home/rof/.maven/maven/bin:/home/rof/.phantomjs/bin:/home/rof/.casperjs/bin:/home/rof/node_modules/.bin:/home/rof/.config/composer/vendor/bin:/home/rof/
50942 verbose lifecycle :/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/go/bin:/home/rof/.rvm/bin
50943 verbose lifecycle @sentry/cli@1.29.1~install: CWD: /home/path/to/frontend/node_modules/@sentry/cli
50944 silly lifecycle @sentry/cli@1.29.1~install: Args: [ '-c', 'node scripts/install.js' ]
50945 silly lifecycle @sentry/cli@1.29.1~install: Returned: code: 1  signal: null
50946 info lifecycle @sentry/cli@1.29.1~install: Failed to exec install script

50958 verbose stack Error: @sentry/cli@1.29.1 install: `node scripts/install.js`
50958 verbose stack Exit status 1
50958 verbose stack     at EventEmitter.<anonymous> (/home/rof/.nvm/versions/node/v8.9.4/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:285:16)
50958 verbose stack     at emitTwo (events.js:126:13)
50958 verbose stack     at EventEmitter.emit (events.js:214:7)
50958 verbose stack     at ChildProcess.<anonymous> (/home/rof/.nvm/versions/node/v8.9.4/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
50958 verbose stack     at emitTwo (events.js:126:13)
50958 verbose stack     at ChildProcess.emit (events.js:214:7)
50958 verbose stack     at maybeClose (internal/child_process.js:925:16)
50958 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
50959 verbose pkgid @sentry/cli@1.29.1
50960 verbose cwd /home/path/to/frontend
50961 verbose Linux 4.4.0-112-generic
50962 verbose argv "/home/rof/.nvm/versions/node/v8.9.4/bin/node" "/home/rof/.nvm/versions/node/v8.9.4/bin/npm" "install"
50963 verbose node v8.9.4
50964 verbose npm  v5.6.0
50965 error code ELIFECYCLE
50966 error errno 1
50967 error @sentry/cli@1.29.1 install: `node scripts/install.js`
50967 error Exit status 1
50968 error Failed at the @sentry/cli@1.29.1 install script.
50968 error This is probably not a problem with npm. There is likely additional logging output above.
50969 verbose exit [ 1, true ]
```

Using:

* node 8.9.4
* yarn 1.3.2
* npm 5.6.0
* @sentry/cli 1.29.1

Thank you

--- a/js/__tests__/helper.test.js
+++ b/js/__tests__/helper.test.js
@@ -1,10 +1,15 @@
 /* eslint-env jest */
 
+const path = require('path');
 const helper = require('../helper');
 
 const SOURCEMAPS_OPTIONS = require('../releases/options/uploadSourcemaps');
 
 describe('SentryCli helper', () => {
+  beforeEach(() => {
+    helper.mockBinaryPath(path.resolve(__dirname, '../__mocks__/sentry-cli'));
+  });
+
   test('call sentry-cli --version', () => {
     expect.assertions(1);
     return helper

--- a/js/__tests__/index.test.js
+++ b/js/__tests__/index.test.js
@@ -1,9 +1,11 @@
 /* eslint-env jest */
 
+const os = require('os');
 const SentryCli = require('../');
 
 describe('SentryCli', () => {
   test('call getPath', () => {
-    expect(SentryCli.getPath()).toContain('sentry-cli');
+    const pattern = os.platform() === 'win32' ? /sentry-cli.exe$/ : /sentry-cli$/;
+    expect(SentryCli.getPath()).toMatch(pattern);
   });
 });

--- a/js/helper.js
+++ b/js/helper.js
@@ -8,10 +8,19 @@ const childProcess = require('child_process');
  * Absolute path to the sentry-cli binary (platform dependant).
  * @type {string}
  */
-const binaryPath =
+let binaryPath =
   os.platform() === 'win32'
     ? path.resolve(__dirname, '..\\bin\\sentry-cli.exe')
     : path.resolve(__dirname, '../sentry-cli');
+
+/**
+ * Overrides the default binary path with a mock value, useful for testing.
+ *
+ * @param {string} mockPath The new path to the mock sentry-cli binary
+ */
+function mockBinaryPath(mockPath) {
+  binaryPath = mockPath;
+}
 
 /**
  * Converts the given option into a command line args array.
@@ -118,10 +127,6 @@ function prepareCommand(command, schema, options) {
  * @returns {string}
  */
 function getPath() {
-  if (process.env.NODE_ENV === 'test') {
-    return path.resolve(__dirname, '__mocks__/sentry-cli');
-  }
-
   return binaryPath;
 }
 
@@ -159,6 +164,7 @@ function execute(args) {
 }
 
 module.exports = {
+  mockBinaryPath,
   serializeOptions,
   prepareCommand,
   getPath,

--- a/js/helper.js
+++ b/js/helper.js
@@ -8,6 +8,7 @@ const childProcess = require('child_process');
  * Absolute path to the sentry-cli binary (platform dependant).
  * @type {string}
  */
+// istanbul ignore next
 let binaryPath =
   os.platform() === 'win32'
     ? path.resolve(__dirname, '..\\bin\\sentry-cli.exe')
@@ -20,32 +21,6 @@ let binaryPath =
  */
 function mockBinaryPath(mockPath) {
   binaryPath = mockPath;
-}
-
-/**
- * Converts the given option into a command line args array.
- *
- * The value can either be an array of values or a single value. The value(s) will be
- * converted to string. If an array is given, the option name is repeated for each value.
- *
- * @example
- * expect(transformOption('--foo', 'a'))
- *   .toEqual(['--foo', 'a'])
- *
- * @example
- * expect(transformOption('--foo', ['a', 'b']))
- *   .toEqual(['--foo', 'a', '--foo', 'b']);
- *
- * @param {string} option The literal name of the option, including dashes.
- * @param {any[]|any} values One or more values for this option.
- * @returns {string[]} An arguments array that can be passed via command line.
- */
-function transformOption(option, values) {
-  if (Array.isArray(values)) {
-    return values.reduce((acc, value) => acc.concat([option.param, String(value)]), []);
-  }
-
-  return [option.param, String(values)];
 }
 
 /**
@@ -87,7 +62,9 @@ function serializeOptions(schema, options) {
         throw new Error(`${option} should be an array`);
       }
 
-      return newOptions.concat(transformOption(schema[option], paramValue));
+      return newOptions.concat(
+        paramValue.reduce((acc, value) => acc.concat([paramName, String(value)]), [])
+      );
     }
 
     if (paramType === 'boolean' || paramType === 'inverted-boolean') {

